dianatanase/assignment2/Q2_tests.java

```diff
@@ -54,17 +55,40 @@ public void testLCA() {
 		assertEquals(7, Q2.getLCA(3,4, tree));
 		assertEquals(2, Q2.getLCA(2,2, tree));
 		assertEquals(2, Q2.getLCA(1,6, tree));
-		assertEquals(3, Q2.getLCA(1,5, tree));
 		assertEquals(2, Q2.getLCA(1,2, tree));
+		assertEquals(3, Q2.getLCA(1,5, tree));
 		assertEquals(3, Q2.getLCA(3,5, tree));
 		assertEquals(3, Q2.getLCA(2,5, tree));
-		
-		// Test with key that are not accepted (do not exist in the tree).
-		assertEquals(-1, Q2.getLCA(7,15, tree));
-		assertEquals(-1, Q2.getLCA(31,7, tree));
-		assertEquals(-1, Q2.getLCA(17,2, tree));
-		assertEquals(-1, Q2.getLCA(5,20, tree));
-
 	}
-
+	
+	// Test with key that are not accepted (do not exist in the tree).		
+	@Test
+	public void testFailingLCA() throws Exception {
+		
+		Tree<Character> tree = setUp();
+		
+		try {
+			 Q2.getLCA(31,7, tree);
+			 fail("Method should fail when key1 does not exist in tree.");
+		} catch (Exception e) {
```

Comment:
You can actually have a test expect an exception
`@Test(expected = IllegalArgumentException.class)`
Then you don't need the fail("reason")s or the try blocks.